--- gnome-system-monitor-3.18.2.orig/src/interface.cpp
+++ gnome-system-monitor-3.18.2/src/interface.cpp
@@ -641,7 +641,7 @@ create_main_window (GsmApplication *app)
     app->main_window = main_window;
 
     session = g_getenv ("XDG_CURRENT_DESKTOP");
-    if (session && !strstr (session, "GNOME")){
+    if ( (!session) || !(strstr (session, "GNOME") || strstr (session, "Deepin"))){
         GtkWidget *mainbox;
         GtkWidget *headerbar;
 
