--- pulseaudio-7.0.orig/src/daemon/default.pa.in
+++ pulseaudio-7.0/src/daemon/default.pa.in
@@ -147,6 +147,11 @@ load-module module-intended-roles
 ### Automatically suspend sinks/sources that become idle for too long
 load-module module-suspend-on-idle
 
+### Automatically switch sound device
+.ifexists module-switch-on-connect@PA_SOEXT@
+load-module module-switch-on-connect
+.endif
+
 ### If autoexit on idle is enabled we want to make sure we only quit
 ### when no local session needs us anymore.
 .ifexists module-console-kit@PA_SOEXT@
