Description: Notify user to logout after changing system font

Author: Hualet Wang <wangyaohua@deepin.com>

Last-Update: 2018-12-07

--- mate-control-center-1.16.1.orig/capplets/appearance/appearance-themes.c
+++ mate-control-center-1.16.1/capplets/appearance/appearance-themes.c
@@ -47,6 +47,8 @@ enum {
 	TARGET_NS_URL
 };
 
+static gboolean font_changed = FALSE;
+
 static const GtkTargetEntry drop_types[] =
 {
 	{"text/uri-list", 0, TARGET_URI_LIST},
@@ -987,6 +989,13 @@ theme_drag_data_received_cb (GtkWidget *
 
 static void background_or_font_changed(GSettings *settings, gchar *key, AppearanceData* data)
 {
+  if (g_strcmp0(key, GTK_FONT_KEY) == 0
+      || g_strcmp0(key, DOCUMENT_FONT_KEY) == 0
+      || g_strcmp0(key, DESKTOP_FONT_KEY) == 0
+      || g_strcmp0(key, WINDOW_TITLE_FONT_KEY) == 0
+      || g_strcmp0(key, MONOSPACE_FONT_KEY) == 0)
+  {
+    font_changed = TRUE;
+  }
+
 	theme_message_area_update(data);
 }
 
@@ -1184,4 +1193,16 @@ themes_shutdown (AppearanceData *data)
   g_free (data->revert_desktop_font);
   g_free (data->revert_windowtitle_font);
   g_free (data->revert_monospace_font);
+
+  if (font_changed)
+  {
+    GtkWidget *dialog = gtk_message_dialog_new(
+        GTK_WINDOW(appearance_capplet_get_widget(data, "appearance_window")),
+        GTK_DIALOG_MODAL, GTK_MESSAGE_INFO, GTK_BUTTONS_OK,
+        "系统温馨提示字体修改后需重新登录才能有完美的效果体验，请注销后重新登录。");
+    gtk_dialog_run(GTK_DIALOG(dialog));
+    gtk_widget_destroy(dialog);
+
+    font_changed = FALSE;
+  }
 }
