--- fontconfig-2.11.1.orig/conf.d/30-metric-aliases.conf
+++ fontconfig-2.11.1/conf.d/30-metric-aliases.conf
@@ -77,15 +77,12 @@ but in an order preferring similar desig
 	  </default>
 	</alias>
 
-<!--
-     Due to Bug#73291, commented out those lines until the broken font are fixed.
 	<alias binding="same">
 	  <family>TeX Gyre Termes</family>
 	  <default>
 	  <family>Times</family>
 	  </default>
 	</alias>
--->
 
 	<alias binding="same">
 	  <family>Nimbus Mono L</family>
@@ -418,10 +415,7 @@ but in an order preferring similar desig
 	<alias binding="same">
 	  <family>Times</family>
 	  <accept>
-<!--
-     Due to Bug#73291, commented out this line until the broken font are fixed.
 	  <family>TeX Gyre Termes</family>
--->
 	  <family>Nimbus Roman No9 L</family>
 	  </accept>
 	</alias>
